app/controllers/posts_controller.rb

```diff
@@ -1,7 +1,6 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find(params[:user_id])
-    @posts = @user.posts
+    @user = User.includes(posts: %i[comments likes]).find(params[:user_id])
```

Comment:
- [ ] Kindly redo your PR screenshots of the n+1 problem. You only just showed the render of the post index page. You should also display the user load, post load and comment load in the BEFORE and AFTER screenshots.

- [ ] I don't think this is how we resolve the n+1 problem. The elimination is done on the post fetch, not on the user fetch.

  ```
  @user = User.find(params[:user_id])
  @posts = @user.posts.includes(:comments)
  ```